misc/docker-ci/build.mk

```diff
@@ -0,0 +1,19 @@
+DOCKER_HUB_NAMESPACE=h2oserver
+MAKEFILE_DIR=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
+PROJECT_ROOT_DIR=$(realpath $(MAKEFILE_DIR)/../..)
+
+all: docker-login h2o-ci h2o-ci-dtrace
+.PHONY: all
+
+docker-login:
+	docker login
+
+h2o-ci: $(MAKEFILE_DIR)/Dockerfile
+	docker build --tag $(DOCKER_HUB_NAMESPACE)/$@ -f $< "$(PROJECT_ROOT_DIR)"
```

Comment:
Clarification question: how do we prohibit entries within current directory from being incorporated to the docker image? The script has to be run in an empty directory, especially if we are to automate this.

Reply:
It makes sense. `cd` to the project root first.

Also, the other reviewed should be satisfied at 834e3b67d835dc376bef2a019b1f7a804f3fb2a9 except for curl version pining -- I coudn't fix `t/50fastcgi.t`.